MathlibTest/LintStyleTextBased.lean

```diff
@@ -0,0 +1,31 @@
+import Mathlib.Tactic.Linter.TextBased
+import Batteries.Data.List.Basic
+
+section unicodeLinter
+
+open Mathlib.Linter.TextBased
+open Mathlib.Linter.TextBased.UnicodeLinter
+
+-- test parsing back error message in `parse?_errorContext` for variant selector errors
```

Comment:
Can you add an error with the text variant also? (And I'm not sure if this test is really flagging all edge cases: for instance the one about the beginning of the line could also be nice to add. Tests can help us, let's use them where applicable.)